packages/foam-vscode/src/features/navigation-provider.ts

```diff
@@ -125,7 +125,14 @@ export class NavigationProvider
       targetRange = section.range;
     }
     const result: vscode.LocationLink = {
-      originSelectionRange: toVsCodeRange(targetLink.range),
+      originSelectionRange: toVsCodeRange(
+        new vscode.Range(
+          targetLink.range.start.line,
+          targetLink.range.start.character + 2,
+          targetLink.range.end.line,
+          targetLink.range.end.character - 2
+        )
+      ),
```

Comment:
Otherwise pressing Ctrl/Cmd when the cursor is over a wikilink will underline the entire `[[...]]` expression, including delimiters.

Reply:
what about direct links? I think this code is used for both

Reply:
What do you mean? The "Go to definition" feature?

Reply:
`targetLink.range` for a wikilink includes `[[` and `]]`
`targetLink.range` for a direct link includes everything from `[` to `)`

Reply:
Sorry for the delay, I've been very busy moving house. I think you're right, I should check for the wikilink type, but I would like to see how the direct link case looks like, perhaps some visual enhancement is in order too.

Reply:
This doesn't seem to be a problem because AFAICS resource.links only include wikilinks.

Reply:
> AFAICS resource.links only include wikilinks

`resource.links` also contains direct links, am I missing something?

(btw, if we create a new `vscode.Range` we don't need the `toVsCodeRange` function)

Reply:
I didn't dig into this, but in a page with a wikilink and a direct link only the first one was included in resource.links. Nevertheless it wouldn't hurt to defensively check for wikilinks.

Reply:
I've added a check and removed the unnecessary conversion.